crates/kernel/src/tasks/scheduler.rs

```diff
@@ -427,83 +453,49 @@ impl Scheduler {
         Err(VerbProgramFailed(VerbProgramError::DatabaseError))
     }
 
-    pub fn submit_shutdown(
-        &self,
-        task: TaskId,
-        reason: Option<String>,
-    ) -> Result<(), SchedulerError> {
-        // If we can't deliver a shutdown message, that's really a cause for panic!
-        self.control_sender
-            .send((task, SchedulerControlMsg::Shutdown(reason)))
-            .expect("could not send clean shutdown message");
-        Ok(())
-    }
-
-    pub fn abort_player_tasks(&self, player: Objid) -> Result<(), SchedulerError> {
-        let mut to_abort = Vec::new();
-        let mut inner = self.inner.write().unwrap();
-        for (task_id, task_ref) in inner.tasks.iter() {
-            if task_ref.player == player {
-                to_abort.push(*task_id);
-            }
-        }
-        for task_id in to_abort {
-            let task = inner.tasks.get_mut(&task_id).expect("Corrupt task list");
-            let tcs = task.task_control_sender.clone();
-            if let Err(e) = tcs.send(TaskControlMsg::Abort) {
-                warn!(task_id, error = ?e, "Could not send abort for task. Dead?");
-                continue;
-            }
-        }
-
-        Ok(())
-    }
-
-    /// Request information on all tasks known to the scheduler.
-    pub fn tasks(&self) -> Result<Vec<TaskDescription>, SchedulerError> {
-        let inner = self.inner.read().unwrap();
+    /// Request information on all (suspended) tasks known to the scheduler.
+    pub fn tasks(&self) -> Vec<TaskDescription> {
+        let inner = self.task_q.lock().unwrap();
```

Comment:
I think this only reads, right? So technically `task_q` could be an `Arc<RwLock<_>>`, and this could be a `.read()`. Does that ... matter?

Reply:
Yeah, RwLock was giving me problems with Send/Sync that Mutex wasn't. I can look into it again in the future.